### ISSUE TITLE
Date field reset glitch when changing part of the date (ex: the month number)

### ISSUE THREAD

**Issue by nekohayo** [invoiceninja/ui#1969]
Similar to #1968 and related to #1361, this time with the date field going wonky:

https://github.com/user-attachments/assets/c095031b-1763-4a43-bf7f-c92063b65774

**Comment by acerspyro** [comment]
Are these fields unit tested?

**Comment by turbo124** [comment]
@nekohayo thanks for this, can you remember which key you hit which caused the start date to reset like that? 

cc
@beganovich @Civolilah

**Comment by nekohayo** [comment]
From memory: I use `Tab` and `Shift+Tab` to navigate forwards/backwards between the fields' sections or the fields, and I typed the month (or year) numbers directly with the keyboard. I don't think I pressed `Enter` anywhere, but I do press `Tab` a lot, and also try to use the `Delete` (and maybe `Backspace`) keys to clear sections within a field. I think it does not like the `Delete` keys and such. I bet you'd be able to encounter these strange behaviors fairly easily by using those keyboard shortcuts in various places :)
